### PR TITLE
BFD-2842: colima compatibility and images in pom

### PR DESCRIPTION
**Pull request by brianburton**
**JIRA Ticket:**
[BFD-2842](https://jira.cms.gov/browse/BFD-2842)

**User Story or Bug Summary:**

As a BFD engineer I want tests to be easily configured and to support test containers on docker platforms other than docker desktop.

---

Running tests under colima (instead of Docker Desktop) failed for a variety of reasons.  This PR makes code changes necessary to allow the tests to pass in that environment.

* Localstack 2.0.2 was not compatible so we now use 2.2.0
* Localstack FIFO queues are broken either in 2.2.0 or when running in colima.  They aren't really necessary for the migrator so SqsDao now uses normal queues.  This simplifies sending messages but complicates receiving them since order isn't guaranteed so tests now account for any possible ordering.
* Tests were failing due to a race condition caused by the postgresql container initializing more slowly under colima.  We now use `waitFor` to cause tests to wait for the container to begin listening to its port before proceeding.

In addition the image to use for localstack is now defined in the root `pom.xml` file.  The class that uses it has a default value in case the property is not defined when a test runs (e.g. in IDEA).  Comments were added to the pom for both image properties to identify where these defaults are defined.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] The data dictionary has been updated with any field mapping changes, if any were made.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-db-migrator/src/test/java/gov/cms/bfd/migrator/app/SqsDaoIT.java [hunk 1]
```diff
@@ -29,24 +31,27 @@ void setUp() {
...
+    String queueName = "my-test-queue";
+    String queueUri = dao.createQueue(queueName);
     String message1 = "this is a first message";
     String message2 = "this is a second message";
-    dao.sendMessage(queueUri, messageGroupId, "1", message1);
-    dao.sendMessage(queueUri, messageGroupId, "2", message2);
-    assertEquals(Optional.of(message1), dao.nextMessage(queueUri));
-    assertEquals(Optional.of(message2), dao.nextMessage(queueUri));
+    dao.sendMessage(queueUri, message1);
+    dao.sendMessage(queueUri, message2);
```
Comment (mackec1) [inline apps/bfd-db-migrator/src/test/java/gov/cms/bfd/migrator/app/SqsDaoIT.java:48]: Could we really try to muck with things by sending message 2 first here?

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/ServerExecutor.java
Comment (mjburling) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/ServerExecutor.java:1]: 👍 Nice little _campsite_ fix here, too. Thanks!

File: apps/bfd-db-migrator/src/main/java/gov/cms/bfd/migrator/app/SqsDao.java
Comment (mjburling) [inline apps/bfd-db-migrator/src/main/java/gov/cms/bfd/migrator/app/SqsDao.java:1]: OK! These changes should yield a little less work for putting the native signaling to work, too.

**Review by mjburling — APPROVED** [review]
✨

**Review by mackec1 — APPROVED** [review]

**Pull request merged**